### ISSUE TITLE
Erroneous results generated, when invalid start_year and end_year are specified

### ISSUE THREAD

**Issue by chengzhuzhang** [E3SM-Project/e3sm_diags#246]
Using [this](https://compy-dtn.pnl.gov/zhan429/run_with_api/multiple_sets_invalidyear/viewer/) run as an example :
```
import os
from acme_diags.parameter.core_parameter import CoreParameter
from acme_diags.parameter.area_mean_time_series_parameter import AreaMeanTimeSeriesParameter
from acme_diags.run import runner

param = CoreParameter()

param.reference_data_path = '/compyfs/e3sm_diags_data/obs_for_e3sm_diags/time-series/'
param.test_data_path = '/compyfs/e3sm_diags_data/test_model_data_for_acme_diags/time-series/E3SM_v1/'

param.test_name = 'e3sm_v1'
param.seasons = ["ANN"]  #seasons shouldn't be a parameter for area_mean_time_series, need to investigate while uncomment this line won't work
param.test_timeseries_input = True
param.test_start_yr = '1998'
param.test_end_yr = '2004'
param.ref_timeseries_input = True
param.ref_start_yr = '1998'
param.ref_end_yr = '2004'

param.multiprocessing = True
param.num_workers = 40
prefix = '/compyfs/www/zhan429/run_with_api'
param.results_dir = os.path.join(prefix, 'multiple_sets_invalidyear')

# Set specific parameters for new sets
ts_param = AreaMeanTimeSeriesParameter()
ts_param.reference_data_path = '/compyfs/e3sm_diags_data/obs_for_e3sm_diags/time-series/'
ts_param.test_data_path = '/compyfs/e3sm_diags_data/test_model_data_for_acme_diags/time-series/E3SM_v1/'
ts_param.test_name = 'e3sm_v1'
ts_param.test_timeseries_input = True
ts_param.test_start_yr = '1998'
ts_param.test_end_yr = '2004'
ts_param.ref_timeseries_input = True
ts_param.ref_start_yr = '1998'
ts_param.ref_end_yr = '2004'

# We're passing in this new object as well, in
# addtion to the CoreParameter object.
runner.sets_to_run = ['lat_lon', 'area_mean_time_series']
runner.run_diags([param, ts_param])
```
with specific config file `ceres.cfg`

```
[#]
sets = ["lat_lon"]
case_id = "CERES-EBAF-TOA-v2.8"
variables = ["RESTOM"]
ref_name = "ceres_ebaf_toa_v2.8"
reference_name = "CERES-EBAF"
seasons = ["ANN"]
contour_levels = [-120, -100, -80, -60, -40, -20, 0, 20, 40, 60, 80]
diff_levels = [-50, -40, -30, -20, -10, -5, 5, 10, 20, 30, 40, 50]


[#]
sets = ["area_mean_time_series"]
case_id = "RESTOM"
variables = ["RESTOM"]
ref_names = ["ceres_ebaf_toa_v2.8", "ceres_ebaf_toa_v4.0"]
regions = ['global']
```

Where cere_ebaf_toa data only available by whole year from 2001. Should stop and print error msg indicating 'invalid years `

**Comment by chengzhuzhang** [comment]
Fix in 7288327

**Comment by chengzhuzhang** [comment]
Fixed in https://github.com/E3SM-Project/e3sm_diags/pull/248
